enterprise/cypher/morsel-runtime/src/test/scala/org/neo4j/cypher/internal/runtime/vectorized/operators/MergeSortOperatorTest.scala

```diff
@@ -75,6 +94,30 @@ class MergeSortOperatorTest extends CypherFunSuite {
     out.longs should equal(Array(9, 9, 14,86,92))
   }
 
+  test("top on two morsels") {
+    val slot = LongSlot(0, nullable = false, CTNode)
+    val columnOrdering = Seq(Ascending(slot))
+    val info = new SlotConfiguration(mutable.Map("apa" -> slot), 1, 0)
+
+    val long1 = Array[Long](1, 2, 3, 4, 5, 6, 7, 8, 9)
+    val long2 = Array[Long](5, 7, 9, 14, 86, 92)
+    val in1 = new Morsel(long1, Array[AnyValue](), long1.length)
+    val in2 = new Morsel(long2, Array[AnyValue](), long2.length)
+    val out = new Morsel(new Array[Long](5), Array[AnyValue](), 5)
+
+    val operator = new MergeSortOperator(columnOrdering, info, Some(Literal(9)))
+
+    val continuation1 = operator.operate(StartLoopWithEagerData(Array(in1, in2), new Iteration(None)), out, null, QueryState(VirtualValues.EMPTY_MAP, null) )
+    continuation1 shouldBe a[ContinueWithSource[_]]
+    out.longs should equal(Array(1, 2, 3, 4, 5))
```

Comment:
This test does not show that `in2` was considered before returning the first morsel. Maybe alternating integers between the inputs would demonstrate the behavior better.